msal/src/Microsoft.Identity.Client/Internal/ModuleInitializer.cs

```diff
@@ -56,6 +56,7 @@ public static void EnsureModuleInitialized()
                 if (!isInitialized)
                 {
                     CoreExceptionFactory.Instance = new MsalExceptionFactory();
+                    CoreTelemetry.Instance = new MSALTelemetry();
```

Comment:
Hmm, this is possibly a bug - you are creating a new instance of MSALTelemetry and also a new instance in the MSALTelemetry class. So in fact you have 2 singletons - one in MSAL and one in core for the same object. 

I see 2 ways out of this: 

1. As stated above, refactor MSAL/ADAL Telemetry objects to not be singletons
2. or initialize CoreTelemetry.Instance with MsalTelemetry.Singleton

Reply:
#resolved with option 2